### PR TITLE
Issues/2035

### LINKED ISSUE #2035: Show unread notification for new screen captures.
Part of #1968. Spec is also available there.

### PR DESCRIPTION
**Pull request by mTwTm**
Fixes #2035

### CODE REVIEW

**Comment by mTwTm** [comment]
DO NOT MERGE: The visual design is not yet ready.

However, the architecture is ready for review.

File: app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java [hunk 1]
```diff
@@ -212,6 +214,7 @@ protected Void doInBackground(Void... voids) {
         map.put(FirebaseHelper.RATE_APP_DIALOG_THRESHOLD, DialogUtils.APP_CREATE_THRESHOLD_FOR_RATE_DIALOG);
         map.put(FirebaseHelper.RATE_APP_NOTIFICATION_THRESHOLD, DialogUtils.APP_CREATE_THRESHOLD_FOR_RATE_NOTIFICATION);
         map.put(FirebaseHelper.SHARE_APP_DIALOG_THRESHOLD, DialogUtils.APP_CREATE_THRESHOLD_FOR_SHARE_DIALOG);
+        map.put(FirebaseHelper.ENABLE_MY_SHOT_UNREAD, MainActivity.ENABLE_MY_SHOT_UNREAD_DEFAULT);
```
Comment (cnevinc) [inline app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java]: Maybe put ENABLE_MY_SHOT_UNREAD_DEFAULT in FirebaseHelper

File: app/src/main/java/org/mozilla/focus/screenshot/CaptureRunnable.java [hunk 2]
```diff
@@ -73,7 +74,11 @@ protected void onPostExecute(final String path) {
             cancel(true);
             return;
         }
-        final int captureResultResource = TextUtils.isEmpty(path) ? R.string.screenshot_failed : R.string.screenshot_saved;
+        boolean captureSuccess = !TextUtils.isEmpty(path);
```
Comment (cnevinc) [inline app/src/main/java/org/mozilla/focus/screenshot/CaptureRunnable.java]: nit: final?

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.java [hunk 3]
```diff
@@ -426,9 +429,11 @@ private void showMenu() {
     }
 
     private void updateMenu() {
+        boolean isMyShotUnreadEnabled = AppConfigWrapper.getMyshotUnreadEnabled(this);
+        boolean showUnread = isMyShotUnreadEnabled && Settings.getInstance(this).hasUnreadMyShot();
```
Comment (cnevinc) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.java]: nit: final?

**Comment by cnevinc** [comment]
There's another FirebaseWrapper under firebase module firebase_no_op build type.
You also need to add getRcBoolean there

File: app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java (hunk 1 above)
Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java]: Do you think I should also move the DialogUtils. ones?

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.java (hunk 3 above)
Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/activity/MainActivity.java]: I'm king of not adding final 😈  Thanks!!

File: app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java (hunk 1 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java]: Maybe I'll move them in the future.

**Pull request merged**